.github/workflows/prepare-release.yml

```diff
@@ -0,0 +1,146 @@
+name: "Prepare CodeQL Coding Standards release"
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: |
+          The version to release (MUST follow semantic versioning).
+        required: true
+      ref:
+        description: |
+          The git commit, branch, or tag to release from.
+        required: true
+
+permissions:
+  contents: write
+  pull-requests: write
+  actions: write
+  checks: write
+
+env:
+  RELEASE_VERSION: ${{ inputs.version }}
+
+jobs:
+  prepare-release:
+    outputs:
+      pull-request-head-sha: ${{ steps.determine-pr-head-sha.outputs.pull-request-head-sha }}
+    name: "Prepare release"
+    if: github.event_name == 'workflow_dispatch'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.ref }}
+
+      - name: Validate release precondition
+        env:
+          RELEASE_VERSION: ${{ inputs.version }}
+          GITHUB_TOKEN: ${{ github.token }}
+        run: |
+          read -r release type < <(gh release list | awk -v release="v$RELEASE_VERSION" '$1 ~ release { print $1,$2; ++n } END { if (n == 0) print "undefined", "undefined" }')
```

Comment:
do you mean to have a gap between `< <` here?

Reply:
Yes, this is a combination of [process substitution](https://www.gnu.org/software/bash/manual/html_node/Process-Substitution.html) and [input redirection](https://www.gnu.org/software/bash/manual/html_node/Redirections.html#Redirecting-Input). 

I guess alternative would be `echo $(...) | read -r -release type`